G/GDAL/build_tarballs.jl

```diff
@@ -3,27 +3,26 @@
 using BinaryBuilder, Pkg
 
 name = "GDAL"
-upstream_version = v"3.6.2"
-version_offset = v"1.0.0"
```

Comment:
Sorry I missed this in review, but this version offset should not have been set back to 0, since now the major version is v300, while the previous one was v301. Should we yank and create a new v301 build?

Reply:
Probably good, yeah